File: packages/EvoHtmlToPdf-v4.12/Samples/NET_4.0/CS/WinForms/EvoHtmlToPdfDemo/SvgToPdfDemo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using EvoPdf;

namespace EvoHtmlToPdfDemo
{
    public partial class SvgToPdf : Form
    {
        public SvgToPdf()
        {
            InitializeComponent();
        }

        private void btnCreatePdf_Click(object sender, EventArgs e)
        {
            this.Cursor = Cursors.WaitCursor;

            try
            {
                PdfConverter pdfConverter = new PdfConverter();

                // set the license key
                pdfConverter.LicenseKey = "B4mYiJubiJiInIaYiJuZhpmahpGRkZE=";

                // We convert the HTML page as local disk file because IIS and VS WebDev Server 
                // don't have .svg included in MIME types by default.
                // When converting HTML pages from an IIS server the MIME type image/svg+xml 
                // must be added to IIS in order to serve the .svg files referenced in HTML
                string htmlWithSvgFilePath = System.IO.Path.Combine(Application.StartupPath, @"..\..\SVG\svg.html");

                // SVG can be used inline in a XML document
                string xmlWithInlineSvgFilePath = System.IO.Path.Combine(Application.StartupPath, @"..\..\SVG\svginline.xml");

                this.Cursor = Cursors.WaitCursor;

                // save the PDF bytes in a file on disk
                string outFilePath = System.IO.Path.Combine(Application.StartupPath, "SvgToPdf.pdf");

                try
                {
                    // call the converter
                    byte[] pdfBytes = pdfConverter.GetPdfBytesFromUrl(htmlWithSvgFilePath);
                    System.IO.File.WriteAllBytes(outFilePath, pdfBytes);
                }
                finally
                {
                    this.Cursor = Cursors.Arrow;
                }

                // open the generated PDF document in an external viewer
                DialogResult dr = MessageBox.Show("Open the rendered file in an external viewer?", "Open Rendered File", MessageBoxButtons.YesNo);
                if (dr == DialogResult.Yes)
                {
                    System.Diagnostics.Process.Start(outFilePath);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }
            finally
            {
                this.Cursor = Cursors.Arrow;
            }
        }

        private void lnkViewSvg_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            // the URL of the HTML with SVG document to convert
            string htmlWithSvgFilePath = System.IO.Path.Combine(Application.StartupPath, @"..\..\SVG\svg.html");
            try
            {
                System.Diagnostics.Process.Start(htmlWithSvgFilePath);
            }
            catch
            {
                MessageBox.Show("There is no HTML viewer installed.");
            }
        }
    }
}

File: packages/EvoHtmlToPdf-v4.12/Samples/NET_2.0/VB/WinForms/EvoHtmlToPdfDemo/SvgToPdfDemo.vb
Imports System.ComponentModel
Imports System.Text

Imports EvoPdf

Namespace EvoHtmlToPdfDemo
	Partial Public Class SvgToPdf
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnCreatePdf_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCreatePdf.Click
			Me.Cursor = Cursors.WaitCursor

			Try
				Dim pdfConverter As New PdfConverter()

				' set the license key
                pdfConverter.LicenseKey = "B4mYiJubiJiInIaYiJuZhpmahpGRkZE="

				' We convert the HTML page as local disk file because IIS and VS WebDev Server 
				' don't have .svg included in MIME types by default.
				' When converting HTML pages from an IIS server the MIME type image/svg+xml 
				' must be added to IIS in order to serve the .svg files referenced in HTML
				Dim htmlWithSvgFilePath As String = System.IO.Path.Combine(Application.StartupPath, "..\..\SVG\svg.html")

				' SVG can be used inline in a XML document
				Dim xmlWithInlineSvgFilePath As String = System.IO.Path.Combine(Application.StartupPath, "..\..\SVG\svginline.xml")

				Me.Cursor = Cursors.WaitCursor

				' save the PDF bytes in a file on disk
				Dim outFilePath As String = System.IO.Path.Combine(Application.StartupPath, "SvgToPdf.pdf")

				Try
					' call the converter
					Dim pdfBytes() As Byte = pdfConverter.GetPdfBytesFromUrl(htmlWithSvgFilePath)
					System.IO.File.WriteAllBytes(outFilePath, pdfBytes)
				Finally
					Me.Cursor = Cursors.Arrow
				End Try

				' open the generated PDF document in an external viewer
				Dim dr As DialogResult = MessageBox.Show("Open the rendered file in an external viewer?", "Open Rendered File", MessageBoxButtons.YesNo)
				If dr = DialogResult.Yes Then
					Process.Start(outFilePath)
				End If
			Catch ex As Exception
				MessageBox.Show(ex.Message)
				Return
			Finally
				Me.Cursor = Cursors.Arrow
			End Try
		End Sub

		Private Sub lnkViewSvg_LinkClicked(ByVal sender As Object, ByVal e As LinkLabelLinkClickedEventArgs) Handles lnkViewSvg.LinkClicked
			' the URL of the HTML with SVG document to convert
			Dim htmlWithSvgFilePath As String = System.IO.Path.Combine(Application.StartupPath, "..\..\SVG\svg.html")
			Try
				Process.Start(htmlWithSvgFilePath)
			Catch
				MessageBox.Show("There is no HTML viewer installed.")
			End Try
		End Sub
	End Class
End Namespace